pkg/vex/statement.go

```diff
@@ -16,25 +16,30 @@ import (
 // [vul_id] for one or more [product_id]s. A VEX Statement exists within a VEX
 // Document.
 type Statement struct {
+	// ID is an optional identifier for the statement. It takes an IRI and must
+	// be unique for each statement in the document.
+	ID string `json:"@id,omitempty"`
```

Comment:
would that be a convention of the IRI for a statement within a document? Or would one have to have uniqueness through the tuple of the parent document and the statement ID? 

It would be nice to have it be a postfix kind of like SPDX namespace + elements.

Reply:
When defined, the IRI must be unique. Not only for the document but universally. Relative IRIs give you that "namespace" effect:

For example, supposing a document is retrieved from the document node id IRI, both statements here would have the same identifier:

```json
{
  "@context": "https://openvex.dev/ns/0.1.0",
  "@id": "https://openvex.dev/docs/public/mydoc/",
  "author": "Unknown Author",
  "timestamp": "2023-07-13T18:36:48.978656084-06:00",
  "version": 1,
  "statements": [
    { "@id": "https://openvex.dev/docs/public/mydoc/mystatement" } ,
    { "@id": "mystatement" } 
  ]
}
```

Reply:
gotcha - sounds good, would it be helpful to add a requirement that the statment IDs should have the prefix of the document IDs?

Reply:
I don't think we should. Statements can use the short form IRI's but may also be an IRI of a statement living in another document, even potentially in another format. While we're not supporting these use cases in our tooling just now, I think we should not break the underlying json-ld fundamentals.